packages/devtools_app/lib/src/screens/profiler/cpu_profile_service.dart

```diff
@@ -19,10 +20,42 @@ extension CpuProfilerExtension on VmServiceWrapper {
       startMicros,
       extentMicros,
     );
-    return CpuProfileData.generateFromCpuSamples(
+
+    const kSamples = 'samples';
+    const kCodeStack = '_codeStack';
+
+    final rawSamples =
+        (cpuSamples.json![kSamples] as List).cast<Map<String, dynamic>>();
+
+    bool buildCodeProfile = false;
+    // If the samples contain a code stack, we should attach them to the
+    // `CpuSample` objects.
+    if (rawSamples.isNotEmpty && rawSamples.first.containsKey(kCodeStack)) {
+      buildCodeProfile = true;
+      final samples = cpuSamples.samples!;
+      for (int i = 0; i < samples.length; ++i) {
+        final cpuSample = samples[i];
+        final rawSample = rawSamples[i];
+        cpuSample.setCodeStack(rawSample[kCodeStack].cast<int>());
+      }
+    }
+
+    final functionProfile = await CpuProfileData.generateFromCpuSamples(
       isolateId: isolateId,
       cpuSamples: cpuSamples,
     );
+    CpuProfileData? codeProfile;
+    if (buildCodeProfile) {
```

Comment:
should we only be doing this in vm developer mode?

Reply:
This is only done in VM developer mode as `getCpuSamples` in `vm_service_wrapper.dart` only requests the `CpuSamples` object with code profile support when VM developer mode is enabled. I can add a comment here explaining that.